### DISCUSSION TITLE
PWM missing from machine

### DISCUSSION THREAD

**Discussion by robhamerling** [micropython/micropython#11586] [STM32 / Pyboard]
I intended to use PWM with a PyBoard 1.1, but PWM is missing from machine (in 1.20.0-93). Is this as designed or a bug?

**Comment by rkompass** [comment]
On the Pyboard PWM is provided through the timers of the pyb module:
Example from the [docs](https://docs.micropython.org/en/latest/pyboard/quickref.html#pwm-pulse-width-modulation):
```
from pyb import Pin, Timer

p = Pin('X1') # X1 has TIM2, CH1
tim = Timer(2, freq=1000)
ch = tim.channel(1, Timer.PWM, pin=p)
ch.pulse_width_percent(50)
```
In the long run this may change and there will be a PWM class in machine.

**Reply by robhamerling** [reply]
Thanks, indeed I found it in the QuickRef for PyBoard!
In DOCS -> Micropython Libraries -> machine -> class PWM  there is no exception mentioned for PyBoard. And since I used machine.PWM with ESP32 and RP2 I was surprised about not being able to use it with PyBoard.
